[Bug] Build fail when BUILD_SPARSE=ON

## 🐛 Bug
Recently, we cannot build from source together with dgl-sparse lib. 
I believe this is due to this PR: https://github.com/dmlc/dgl/pull/4877, where it removes some macro defs probably suffered from multiple definition as here: https://github.com/dmlc/dgl/blob/c8ea9fa4e4afd3d16d86afa99494095060685c10/dgl_sparse/include/sparse/dgl_headers.h#L20-L40

However, the problem is, if users build backend pytorch with flag `USE_GLOG=OFF`, then the DGL build will fail because many check ops are not defined in `c10/util/logging_is_not_google_glog.h`. 

I am wondering why not just replace all check ops in `src/sparse_matrix.cc` to `TORCH_CHECK_OPS` to avoid any confusion?


## To Reproduce

1. Build pytorch with `USE_GLOG=OFF`
2. Build DGL with `BUILD_SPARSE=ON`

Err msg:
```
/workspace/dgl/dgl_sparse/src/sparse_matrix.cc: In constructor 'dgl::sparse::SparseMatrix::SparseMatrix(const std::shared_ptr<dgl::sparse::COO>&, const std::shared_ptr<dgl::sparse::CSR>&, const std::shared_ptr<dgl::sparse::CSR>&, at::Tensor, const std::vector<long int>&)':
/workspace/dgl/dgl_sparse/src/sparse_matrix.cc:26:3: error: 'CHECK_EQ' was not declared in this scope; did you mean 'CHECK_IF'?
   26 |   CHECK_EQ(shape.size(), 2)
      |   ^~~~~~~~
      |   CHECK_IF
```


## Environment

 - DGL Version (e.g., 1.0): 1.0
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3): 1.13
 - OS (e.g., Linux): ubuntu
 - How you installed DGL (`conda`, `pip`, source): source

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -36,6 +36,8 @@ dgl_option(USE_HDFS "Build with HDFS support" OFF) # Set env HADOOP_HDFS_HOME if
 dgl_option(REBUILD_LIBXSMM "Clean LIBXSMM build cache at every build" OFF) # Set env HADOOP_HDFS_HOME if needed
 dgl_option(USE_EPOLL "Build with epoll for socket communicator" ON)
 dgl_option(TP_BUILD_LIBUV "Build libuv together with tensorpipe (only impacts Linux)" ON)
+dgl_option(BUILD_SPARSE "Build DGL sparse library" OFF)
+dgl_option(TORCH_PYTHON_INTERPS "Python interpreter used to build tensoradapter and DGL sparse library" python3)
 
 # Set debug compile option for gdb, only happens when -DCMAKE_BUILD_TYPE=DEBUG
 if (NOT MSVC)

--- a/dgl_sparse/CMakeLists.txt
+++ b/dgl_sparse/CMakeLists.txt
@@ -13,6 +13,9 @@ message(STATUS "find_cmake.py output: ${TORCH_PREFIX_VER}")
 list(GET TORCH_PREFIX_VER 0 TORCH_PREFIX)
 list(GET TORCH_PREFIX_VER 1 TORCH_VER)
 message(STATUS "Configuring for PyTorch ${TORCH_VER}")
+string(REPLACE "." ";" TORCH_VERSION_LIST ${TORCH_VER})
+list(GET TORCH_VERSION_LIST 0 TORCH_VERSION_MAJOR)
+list(GET TORCH_VERSION_LIST 1 TORCH_VERSION_MINOR)
 
 if(USE_CUDA)
   add_definitions(-DDGL_USE_CUDA)
@@ -34,6 +37,8 @@ add_library(${LIB_DGL_SPARSE_NAME} SHARED ${SPARSE_SRC} ${SPARSE_HEADERS})
 target_include_directories(
   ${LIB_DGL_SPARSE_NAME} PRIVATE ${SPARSE_DIR} ${SPARSE_HEADERS})
 target_link_libraries(${LIB_DGL_SPARSE_NAME} "${TORCH_LIBRARIES}")
+target_compile_definitions(${LIB_DGL_SPARSE_NAME} PRIVATE TORCH_VERSION_MAJOR=${TORCH_VERSION_MAJOR})
+target_compile_definitions(${LIB_DGL_SPARSE_NAME} PRIVATE TORCH_VERSION_MINOR=${TORCH_VERSION_MINOR})
 
 target_include_directories(${LIB_DGL_SPARSE_NAME} PRIVATE "${CMAKE_SOURCE_DIR}/third_party/dmlc-core/include")
 target_link_libraries(${LIB_DGL_SPARSE_NAME} dmlc)

--- a/dgl_sparse/include/sparse/dgl_headers.h
+++ b/dgl_sparse/include/sparse/dgl_headers.h
@@ -39,4 +39,14 @@
 #undef DLOG
 #undef LOG_IF
 
+// For Pytorch version later than 1.12, redefine CHECK_* to TORCH_CHECK_*.
+#if !(TORCH_VERSION_MAJOR == 1 && TORCH_VERSION_MINOR <= 12)
+#define CHECK_EQ(val1, val2) TORCH_CHECK_EQ(val1, val2)
+#define CHECK_NE(val1, val2) TORCH_CHECK_NE(val1, val2)
+#define CHECK_LE(val1, val2) TORCH_CHECK_LE(val1, val2)
+#define CHECK_LT(val1, val2) TORCH_CHECK_LT(val1, val2)
+#define CHECK_GE(val1, val2) TORCH_CHECK_GE(val1, val2)
+#define CHECK_GT(val1, val2) TORCH_CHECK_GT(val1, val2)
+#endif
+
 #endif  // SPARSE_DGL_HEADERS_H_

